utils/smtp_client.py

```diff
@@ -11,137 +11,81 @@
-    smtp_config = get_smtp_config(smtp_secret_path, settings)
-    host = smtp_config['server']
-    port = smtp_config['port']
-    _username = smtp_config['username']
-    password = smtp_config['password']
-    _mail_address = config['smtp']['mail_address']
-
-    return init(host, port, _username, password, client_only=client_only)
-
-
-def get_smtp_config(path, settings):
-    config = {}
-
-    required_keys = ('password', 'port', 'require_tls', 'server', 'username')
-    secret_reader = SecretReader(settings=settings)
-    data = secret_reader.read_all({'path': path})
-
-    try:
-        for k in required_keys:
-            config[k] = data[k]
-    except KeyError as e:
-        raise Exception("Missing expected SMTP config key in vault secret: {}"
-                        .format(e))
-
-    return config
-
-
-def get_mails(folder='INBOX', criteria='ALL', settings=None):
-    global _server
-
-    if _server is None:
-        init_from_config(settings, client_only=False)
-
-    _server.select(f'"{folder}"')
-
-    result, data = _server.uid('search', None, criteria)
-    uids = [s for s in data[0].split()]
-    results = []
-    for uid in uids:
-        result, data = _server.uid('fetch', uid, '(RFC822)')
-        msg = data[0][1].decode('utf-8')
-        results.append({'uid': uid, 'msg': msg})
-
-    return results
-
-
-@retry()
-def send_mail(names, subject, body, settings=None):
-    global _client
-    global _username
-    global _mail_address
-
-    if _client is None:
-        init_from_config(settings)
-
-    msg = MIMEMultipart()
-    from_name = str(Header('App SRE team automation', 'utf-8'))
-    msg['From'] = formataddr((from_name, _username))
-    to = set()
-    for name in names:
-        if '@' in name:
-            to.add(name)
-        else:
-            to.add(f"{name}@{_mail_address}")
-    msg['To'] = ', '.join(to)
-    msg['Subject'] = subject
-
-    # add in the message body
-    msg.attach(MIMEText(body, 'plain'))
-
-    # send the message via the server set up earlier.
-    _client.sendmail(_username, to, msg.as_string())
-
-
-def send_mails(mails, settings=None):
-    global _client
-
-    init_from_config(settings)
-    try:
+class SmtpClient:
+    def __init__(self, settings, client_only=True):
+        config = get_config()
+
+        smtp_secret_path = config['smtp']['secret_path']
+        smtp_config = self.get_smtp_config(smtp_secret_path, settings)
+        self.host = smtp_config['server']
+        self.port = str(smtp_config['port'])
+        self.user = smtp_config['username']
+        self.passwd = smtp_config['password']
+        self.mail_address = config['smtp']['mail_address']
+
+        self.client = smtplib.SMTP(host=self.host,
+                                   port=self.port)
+        self.client.send
+        self.client.starttls()
+        self.client.login(self.user, self.passwd)
+
+        self.server = None
+        if not client_only:
+            self.server = self._init_server()
```

Comment:
this should actually be a lazy init. no reason to have the client_only path.

Reply:
will update the PR with that

Reply:
you updated the PR just before I was going to suggest it 😂  Great!